### ISSUE TITLE
proposal period with short id

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#176]
I've run  `proposal period 472f2420820c9933` and got:
```
  File "~/.local/bin/pacli", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/__main__.py", line 479, in main
    fire.Fire({
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 143, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 477, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/fire/core.py", line 693, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/dt_classes.py", line 605, in period
    return self.__current_period(proposal, blockheight=blockheight, show_blockheights=True, mode=mode, debug=debug)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/dt_classes.py", line 504, in __current_period
    proposal_id = eu.search_for_stored_tx_label("proposal", proposal, quiet=quiet)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.local/lib/python3.12/site-packages/pacli/extended_utils.py", line 480, in search_for_stored_tx_label
    raise ei.PacliInputDataError("Label '{}' not found.".format(identifier))
pacli.extended_interface.PacliInputDataError: Label '472f2420820c9933' not found
```
The same command with the full proposal's id (`proposal period 472f2420820c99333969bba0ecb7fc4080b7e2f6d8bb65756070a31b450b7432
` in this case) works fine.

**Comment by d5000** [comment]
Short ids and searches currently are not implemented in all proposal commands, "period" being one of the commands where it wasn't implemented. The reason is that short ids have to be searched and that takes longer than simply retrieving a transaction (by the complete id) or a label (from the configuration file).

I have it now implemented differently, because I see this can lead to confusions. The `--miniid` (`-m`) flag and the call of proposals by searching in the description will be now available for more commands, including all `proposal show` options, `proposal period` and `proposal voters`.

Commit: 4fa17d6

Can be closed if everything works, I've pretested it.
----

A general comment: In general dPoD testing is not being prioritary now. The dPoD commands are relatively well tested already, and as dPoD tokens should be launched several months after PoB tokens, we can do a final testing round once the PoB/AT "section" is finished. DEX is more prioritary at this moment.

If you want to test the dPoD commands (proposal, donation etc.) anyway, no problem, but expect that I may put several issues on hold and test them when unittests are completed. I will try to solve the issues which are already open though. In order for you to not "waste" time, I think in general it would also be better if you wait for the unittests.

So if there are still any commands left you want to test in the PoB/AT/DEX categories, then I'd ask you to finish testing on them first, so these categories can be "closed" and I can see finaly some light behind the tunnel.

**Comment by buhtignew** [comment]
Do we really need the `-m` flag?
Maybe after having searched among the the trx_ids and labels if nothing found the code can check whether the string is not a mini-id?
If it takes too much people will tend to avoid using the mini-ids at all. 
Another idea could be: do we really need the mini-ids? Since there are labels those who want to use a mini-id can add it as a label.
_______________________________________________
> In general dPoD testing is not being prioritary now. 

I'm aware about it. The dPoD token testing is lower in my priority list than the dex testing as well, but since I was stuck at #174 I've preferred to go on with the lower priority work rather than just doing nothing.
FYI, except for the issues already opened all the groups testing on my side is finished a part `dex`, `proposal`, `donation` and `podtoken`.
What bothers me about the groups I consider to have tested most is that since we've worked hard and the quality standard of the code is quite high what we've tested is not inline with vanilla commands that may contain issues we are not aware of (in the issue #168 there is my proposal about).
However I'd like to re-test `proposal`, `donation` and `podtoken` as well although we've already tested them in the past, firstly because I have a better understanding of the whole picture now and there are commands I haven't tested at all (like `podtoken spawn`, for instance).
And in the second place we've changed the structure of our commands which can have caused some small bug we are not aware about.
In any case if you think it's better for me to wait for the unitests to be completed I'll be waiting for your hint on regard then. So please let me know when I should restart with them.

**Comment by d5000** [comment]
> Do we really need the -m flag?

Originally it was needed due to the issue I explained in issue #180 . You can now search by mini-id without the flag. But to be absolutely sure that nobody for example puts the miniid in the description to confuse or attack a proposal, I prefer to keep the `-m` flag. If you use the flag the miniid must exactly coincide with your input.

I may revise the help strings of the commands which use this kind of proposal search, so this becomes clearer.

The miniid itself is important, as it forms part of the identification string of the proposal, together with the description. The idea is to have an identification which is easier to "grasp" as the complete ID, but cannot be faked easily (to avoid a "duplicate proposal attack"). 

The miniid is more difficult to fake than the description, you'd have to create and sign a lot of proposal txes to match a miniid so you have to waste computational resources, this attack is similar to Vanitygen.

> FYI, excepting the issues already opened, all the groups testing on my side is finished

That's awesome! I'll concentrate then on the issue you mentioned so you can continue with the DEX. The dPoD testing can be paused until unittests are ready.

**Comment by buhtignew** [comment]
I'm closing this since everything is clear for me here.
Thank you!
